### PR TITLE
Imxrt1060 enablement

### PR DESCRIPTION
**Pull request by jhqian**
Enable Zephyr on i.MXRT1060
- Add soc files (header/svd/clock/iomux)
- Add board support for mimxrt1060_evk, the following samples have been tested:
  - hello_world, philosophers, synchronization, basic/blinky, and basic/button.

### CODE REVIEW

**Review by galak — COMMENTED** [review]
Looks general good, a few comments about copyright years being 2017 instead of 2018

**Review by galak — CHANGES_REQUESTED** [review]
In the commit that imports files into ext: we need reference to where things are coming from beyond just MCUXpress SDK 2.4.0.

Also should we just update all the imx-rt support across the board to SDK 2.4.0?

**Comment by jhqian** [comment]
@galak, the SDK can be downloaded from mcuxpresso.nxp.com. Will it work if I put this into the commit log? This set of patches are only to enable RT1060. I was not going to update the whole SDK 2.4.0 to Zephyr.

File: boards/arm/mimxrt1060_evk/Kconfig.board
Comment (jhqian) [reply boards/arm/mimxrt1060_evk/Kconfig.board]: Will update, thanks.

File: arch/arm/soc/nxp_imx/rt/Kconfig.defconfig.mimxrt1062
Comment (jhqian) [reply arch/arm/soc/nxp_imx/rt/Kconfig.defconfig.mimxrt1062]: Will update

**Comment by jhqian** [comment]
@galak , just like I mentioned, these patches were only for the SoC support. But yes, I can grab all the other drivers released along with RT1060 here. Will it be okay I create another PR for the driver update?

**Comment by galak** [comment]
> @galak , just like I mentioned, these patches were only for the SoC support. But yes, I can grab all the other drivers released along with RT1060 here. Will it be okay I create another PR for the driver update?

Sure a new PR for updating to SDK 2.4.0 makes sense, leave it up to you if you want to import the imxrt1060 files as part of that or not.

**Comment by jhqian** [comment]
@galak , thanks. So these patches have been approved for merge, right? I'm working on the new PR for the other drivers.

**Comment by galak** [comment]
> @galak , thanks. So these patches have been approved for merge, right? I'm working on the new PR for the other drivers.

while wait for @dbkinder to review the docs.

**Comment by jhqian** [comment]
okay, thanks.

**Comment by jhqian** [comment]
Just updated the mimxrt1060_evk.rst for the EVK UG link.

File: boards/arm/mimxrt1060_evk/Kconfig.defconfig
Comment (MaureenHelm) [inline boards/arm/mimxrt1060_evk/Kconfig.defconfig]: Remove the configs with `def_bool n`; they are redundant.

File: boards/arm/mimxrt1060_evk/doc/mimxrt1060_evk.rst
Comment (MaureenHelm) [inline boards/arm/mimxrt1060_evk/doc/mimxrt1060_evk.rst]: Remove this link

File: Jenkinsfile
Comment (MaureenHelm) [inline Jenkinsfile]: @hakehuang Why is this file in the PR?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Please update `ext/hal/nxp/mcux/README`

We should include the drivers in this PR (in the same commit as the SoC headers), since they come from the same MCUXpresso SDK distribution.

**Comment by MaureenHelm** [comment]
If #9776 goes in first (and it probably will), then you will need to move the rt1060 device support accordingly.

File: boards/arm/mimxrt1060_evk/Kconfig.defconfig
Comment (jhqian) [reply boards/arm/mimxrt1060_evk/Kconfig.defconfig]: will do

File: boards/arm/mimxrt1060_evk/doc/mimxrt1060_evk.rst
Comment (jhqian) [reply boards/arm/mimxrt1060_evk/doc/mimxrt1060_evk.rst]: ok

File: boards/arm/mimxrt1060_evk/doc/mimxrt1060_evk.rst
Comment (dbkinder) [inline boards/arm/mimxrt1060_evk/doc/mimxrt1060_evk.rst]: delete the blank line and indenting:

    The default configuration can be found in the defconfig file:
    ``boards/arm/mimxrt1060_evk/mimxrt1060_evk_defconfig``.

File: boards/arm/mimxrt1060_evk/doc/mimxrt1060_evk.rst
Comment (dbkinder) [inline boards/arm/mimxrt1060_evk/doc/mimxrt1060_evk.rst]: spelled with a slash ``I/Os``  (and add the extra title underline character)

File: boards/arm/mimxrt1060_evk/doc/mimxrt1060_evk.rst
Comment (dbkinder) [inline boards/arm/mimxrt1060_evk/doc/mimxrt1060_evk.rst]: OK, but how **do** you flash the board?  Sounds like you're assuming the reader will know.

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: boards/arm/mimxrt1060_evk/doc/mimxrt1060_evk.rst
Comment (jhqian) [reply boards/arm/mimxrt1060_evk/doc/mimxrt1060_evk.rst]: it is described in the EVK UG.

File: boards/arm/mimxrt1060_evk/doc/mimxrt1060_evk.rst
Comment (jhqian) [reply boards/arm/mimxrt1060_evk/doc/mimxrt1060_evk.rst]: ok

File: boards/arm/mimxrt1060_evk/doc/mimxrt1060_evk.rst
Comment (jhqian) [reply boards/arm/mimxrt1060_evk/doc/mimxrt1060_evk.rst]: what is "extra title underline character"?

Comment (jhqian) [reply boards/arm/mimxrt1060_evk/doc/mimxrt1060_evk.rst]: @dbkinder , "extra title underline character" has been added. Thanks.

File: boards/arm/mimxrt1060_evk/doc/mimxrt1060_evk.rst
Comment (dbkinder) [inline boards/arm/mimxrt1060_evk/doc/mimxrt1060_evk.rst]: Add after this sentence:

    Instead, see the `MIMXRT1060-EVK User Guide`_ for flashing instructions.

**Review by dbkinder — DISMISSED** [review]

File: boards/arm/mimxrt1060_evk/doc/mimxrt1060_evk.rst
Comment (jhqian) [reply boards/arm/mimxrt1060_evk/doc/mimxrt1060_evk.rst]: > Add after this sentence:
> 
> ```
> Instead, see the `MIMXRT1060-EVK User Guide`_ for flashing instructions.
> ```

Thank you, just checked the online doc, that flash programming thing is described in another application note. I updated the rst file giving the link to that AN for reference.

**Review by MaureenHelm — APPROVED** [review]

**Comment by MaureenHelm** [comment]
@galak @dbkinder ping

File: boards/arm/mimxrt1060_evk/mimxrt1060_evk.dts
Comment (galak) [inline boards/arm/mimxrt1060_evk/mimxrt1060_evk.dts]: led@0 -> led_1

File: boards/arm/mimxrt1060_evk/mimxrt1060_evk.dts
Comment (galak) [inline boards/arm/mimxrt1060_evk/mimxrt1060_evk.dts]: button@0 -> button_1

File: boards/arm/mimxrt1060_evk/mimxrt1060_evk.dts
Comment (galak) [inline boards/arm/mimxrt1060_evk/mimxrt1060_evk.dts]: hyperflash@60000000

File: boards/arm/mimxrt1060_evk/mimxrt1060_evk.dts
Comment (galak) [inline boards/arm/mimxrt1060_evk/mimxrt1060_evk.dts]: qspi@60000000

**Review by galak — DISMISSED** [review]
Some minor DTS comments.

**Comment by MaureenHelm** [comment]
@jhqian Please see #10024 to fix CI failures.

**Comment by galak** [comment]
The commit order seems wrong, in the NXP repo it seems the hal is added last, this should be first.

**Pull request merged**
